### PR TITLE
[Keyboard] Baion 808

### PR DESCRIPTION
**Pull request by yiancar**
Add Baion 808 keyboard:)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/baion_808/config.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 17
+
```
Comment (waffle87) [inline keyboards/baion_808/config.h:25]:
```suggestion
```

File: keyboards/baion_808/config.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * or via bootmagic (hold SPACE+N while plugging in the keyboard). Once set by
+ * bootmagic, NKRO mode will always be enabled until it is toggled again during a
+ * power-up.
+ *
+ */
+//#define FORCE_NKRO
+
+/* Bootmagic Lite key configuration */
+#define BOOTMAGIC_LITE_ROW 0
+#define BOOTMAGIC_LITE_COLUMN 0
```
Comment (waffle87) [inline keyboards/baion_808/config.h:74]:
```suggestion
```

File: keyboards/baion_808/readme.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+* Hardware Availability: https://baionlenja.com 
+
+## Instructions
+
+### Build
+
+Make example for this keyboard (after setting up your build environment):
+
+    make baion_808:default
+
```
Comment (waffle87) [inline keyboards/baion_808/readme.md:16]:
```suggestion
Flashing example for this keyboard:

    make baion_808:default:flash

```

File: keyboards/baion_808/rules.mk [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# MCU name
+MCU = STM32F072
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
```
Comment (waffle87) [inline keyboards/baion_808/rules.mk:6]:
```suggestion
```

File: keyboards/baion_808/info.json [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{
+    "keyboard_name": "Baion 808",
+    "manufacturer": "Yiancar-Designs",
+    "url": "https://yiancar-designs.com",
+    "maintainer": "Yiancar-Designs",
+    "usb": {
+        "vid": "0x8968",
+        "pid": "0x4238",
+        "device_version": "0.0.1"
+    },
```
Comment (waffle87) [inline keyboards/baion_808/info.json:10]:
```suggestion
    },
    "processor": "STM32F072",
    "bootloader": "stm32-dfu",
```

File: keyboards/baion_808/keymaps/default/readme.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# The default keymap for Baion 808. VIA support disabled.
+
+![Layer 0](https://i.imgur.com/IhbvSZ1.png)
```
Comment (waffle87) [inline keyboards/baion_808/keymaps/default/readme.md:3]: Both the keymap `readme.md`'s should be removed and this image displayed in keyboard level `readme.md`, if you feel necessary.

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by yiancar** [comment]
Done!

**Review by waffle87 — APPROVED** [review]

File: keyboards/baion_808/readme.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+![Layer 0](https://i.imgur.com/IhbvSZ1.png)
+
+## Instructions
+
+### Build
+
+Make example for this keyboard (after setting up your build environment):
+
+    make baion_808:default
```
Comment (zvecr) [inline keyboards/baion_808/readme.md:17]:
```suggestion
    make baion_808:default

```

**Review by zvecr — CHANGES_REQUESTED** [review]
Also does the following image represent the PCB?
image

**Comment by yiancar** [comment]
@zvecr indeed so:)

**Comment by zvecr** [comment]
> @zvecr indeed so:)

Then I would expect the layout macro to be `LAYOUT_all` and it should also implement the tlk community layouts.

**Comment by yiancar** [comment]
Can add the tlk community layout.
So in that case the suggestion would be LAYOUT_all for via and then a LAYOUT_tkl_f13_ansi_split_bs_rshift for the comunity?

**Comment by yiancar** [comment]
@zvecr I think now everyone should be happy with the layouts?:)

**Comment by yiancar** [comment]
Bump:)

**Review by fauxpark — APPROVED** [review]

**Comment by yiancar** [comment]
Sorry I made a mistake with f13 here... give me a few to fix that..

**Comment by yiancar** [comment]
Sorry for all the mess, it is now correct...

**Comment by yiancar** [comment]
OK CI is failing on [talljoe-tkl](https://github.com/qmk/qmk_firmware/tree/master/layouts/community/tkl_ansi/talljoe-tkl) this layout :/ I think the layout itself is broken?

**Comment by sigprof** [comment]
Yes, looks like most layouts and some keymaps by @Talljoe are broken since #17229, included in [QMK 0.18.0](https://docs.qmk.fm/#/ChangeLog/20220827). In #17559 (also included in QMK 0.18.0) a workaround for similar cases was added (the source file that actually defines the `keymaps` array needs to be specified in the `INTROSPECTION_KEYMAP_C` make variable and removed from `SRC`).

**Pull request merged**